raiden_contracts/deploy/__main__.py

```diff
@@ -744,99 +729,70 @@ def verify_deployed_service_contracts(
     assert contract_manager.contracts_version == deployment_data['contracts_version']
     assert chain_id == deployment_data['chain_id']
 
+    def service_bundle_checker(deploy_record, onchain_instance):
+        # We need to also check the constructor parameters against the chain
+        constructor_arguments = deploy_record['constructor_arguments']
+        assert to_checksum_address(
+            onchain_instance.functions.token().call(),
+        ) == token_address
+        assert token_address == constructor_arguments[0]
     service_bundle = verify_deployed_contract(
         web3,
         contract_manager,
         deployment_data,
         CONTRACT_SERVICE_REGISTRY,
+        service_bundle_checker,
     )
 
-    # We need to also check the constructor parameters against the chain
-    constructor_arguments = deployment_data['contracts'][
-        CONTRACT_SERVICE_REGISTRY
-    ]['constructor_arguments']
-    assert to_checksum_address(
-        service_bundle.functions.token().call(),
-    ) == token_address
-    assert token_address == constructor_arguments[0]
-
-    print(
-        f'{CONTRACT_SERVICE_REGISTRY} at {service_bundle.address} '
-        f'matches the compiled data from contracts.json',
-    )
-
+    def user_deposit_checker(deploy_record, onchain_instance):
+        # We need to also check the constructor parameters against the chain
+        constructor_arguments = deploy_record['constructor_arguments']
```

Comment:
This line is in all checkers, so `verify_deployed_contract` could call each checker directly with `constructor_arguments`.

Reply:
Yes.

I still wonder why a special treatment is necessary for constructor arguments, but sure, that's the way it is.